src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Await.cs

```diff
@@ -1,15 +1,41 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
 using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed partial class LocalRewriter
     {
         public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
+        {
+            return VisitAwaitExpression(node, true);
+        }
+
+        public BoundExpression VisitAwaitExpression(BoundAwaitExpression node, bool used)
         {
             _sawAwait = true;
-            return base.VisitAwaitExpression(node);
+            var rewrittenAwait = (BoundExpression)base.VisitAwaitExpression(node);
+            if (!used)
+            {
+                // Await expression is already at the statement level.
+                return rewrittenAwait;
+            }
+
+            // The await expression will be lowered to code that involves the use of side-effects
+            // such as jumps and labels, therefore it is represented by a BoundSpillSequence.
+            // The resulting nodes will be "spilled" to move such statements to the top
+            // level (i.e. into the enclosing statement list).
+            _needsSpilling = true;
+            BoundLocal replacement = _factory.StoreToTemp(
```

Comment:
`await` is never an lvalue, so this temp is never a ref temp. 
In general case we can have byref spills. 

In a case of `await`, they will be later decomposed into something that can be hoisted like {array, index} pairs or will produce an error if it is a true byref spill - like a result of byref method call.

In a case when spill is caused by expression switch or stackalloc (in the future), byref spills would work just fine.

Perhaps need a bit more care to make sure that "readonliness" matches. - I.E. if spilling an `in` argument, the temp must be `ref readonly`. Not sure if this is already handled correctly.

Reply:
The part about "readonly" is probably better handled as a separate workitem. It would be a new scenario for spiller. 
In `await` case we decompose ref spills, so readonlyness of temps did not matter since they would not survive.